src/pynwb/file.py

```diff
@@ -610,6 +619,40 @@ def add_stimulus_template(self, timeseries):
         self._add_stimulus_template_internal(timeseries)
         self._update_sweep_table(timeseries)
 
+    @docval({'name': 'data', 'type': (np.array, list, tuple, pd.DataFrame, NWBContainer, ScratchData),
+             'help': 'the data to add to the scratch space'},
+            {'name': 'name', 'type': str,
+             'help': 'the name of the data. Only used when passing in np.array, list, or tuple', 'default': None},
+            {'name': 'notes', 'type': str,
+             'help': 'notes to add to the data. Only used when passing in np.array, list, or tuple', 'default': None},
+            {'name': 'table_description', 'type': str,
+             'help': 'description for table. Only used when passing in pd.DataFrame', 'default': None})
+    def add_scratch(self, **kwargs):
```

Comment:
I think it would be less confusing (though a little redundant) to make name be required for NWBContainers and ScratchData as well. Then the scratch space would act like a dictionary.

Reply:
I would like to minimize boilerplate cruft, and start using reasonable defaults where possible. Making `name` optional does not preclude dictionary like behavior